auction-server/src/auction.rs

```diff
@@ -1066,3 +1034,353 @@ async fn simulate_bid_svm(chain_store: &ChainStoreSvm, bid: &BidSvm) -> Result<(
         None => Ok(()),
     }
 }
+
+pub trait Auction {
+    type Block: DebugTrait;
+    type BlockStream<'a>: Stream<Item = Self::Block> + Unpin + Send + 'a;
+    type WsClient;
+    type SimulatedBid: SimulatedBidTrait + Clone;
```

Comment:
Do we need to specifically mention Clone? SimulatedBidTrait already includes Clone